flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/tasks/CheckpointCoordinatorConfiguration.java

```diff
@@ -63,11 +65,13 @@ public CheckpointCoordinatorConfiguration(
 			int maxConcurrentCheckpoints,
 			CheckpointRetentionPolicy checkpointRetentionPolicy,
 			boolean isExactlyOnce,
-			boolean isPerfetCheckpointForRecovery) {
+			boolean isPerfetCheckpointForRecovery,
+			int tolerableCpFailureNumber) {
 
 		// sanity checks
 		if (checkpointInterval < 1 || checkpointTimeout < 1 ||
-			minPauseBetweenCheckpoints < 0 || maxConcurrentCheckpoints < 1) {
+			minPauseBetweenCheckpoints < 0 || maxConcurrentCheckpoints < 1 ||
+			tolerableCpFailureNumber < 0 || tolerableCpFailureNumber >= Integer.MAX_VALUE) {
```

Comment:
You overlooked to change this check `tolerableCpFailureNumber > Integer.MAX_VALUE` or else a valid config cann trigger exceptions.

Reply:
Sorry, it's my negligence. Have fixed